splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java

```diff
@@ -503,9 +503,10 @@ private void hide(final AppCompatActivity activity, final int fadeOutDuration, b
         // Hide with Android 12 API
         if (null != this.onPreDrawListener) {
             this.isVisible = false;
-            final View content = activity.findViewById(android.R.id.content);
-            content.getViewTreeObserver().removeOnPreDrawListener(this.onPreDrawListener);
             this.onPreDrawListener = null;
+            if (null != content) {
+                content.getViewTreeObserver().removeOnPreDrawListener(this.onPreDrawListener);
```

Comment:
you have moved this line after the `this.onPreDrawListener = null;`, so as `this.onPreDrawListener` is null, `removeOnPreDrawListener` is not having any effect.

Reply:
Good catch 🤦‍♂️